app/Http/Controllers/Manager/TeamController.php

```diff
@@ -0,0 +1,97 @@
+<?php
+
+namespace App\Http\Controllers\Manager;
+
+use Illuminate\Http\Request;
+use App\Models\Team;
+class TeamController extends \App\Http\Controllers\Controller
+{
+    /**
+     * Display a listing of the resource.
+     *
+     * @return \Illuminate\Http\Response
+     */
+    public function index()
+    {
+        $teams = Team::all();
+        return view('manager.teams.index',compact('teams'));
+    }
+
+    /**
+     * Show the form for creating a new resource.
+     *
+     * @return \Illuminate\Http\Response
+     */
+    public function create()
+    {
+        return view('manager.teams.create');
+    }
+
+    /**
+     * Store a newly created resource in storage.
+     *
+     * @param  \Illuminate\Http\Request  $request
+     * @return \Illuminate\Http\Response
+     */
+    public function store(Request $request)
+    {
+        $validated = $request->validate([
+            'name' => 'required|max:20', 
+        ]);
+        $owner_id = auth()->id();
+        $team = new Team($validated);
+        $team->owner_id = $owner_id;
+        $team->save();
+        return to_route('manager.teams.show', $team)->with('success', 'チームを作成しました');
+    }
+
+    /**
+     * Display the specified resource.
+     *
+     * @param  int  $id
+     * @return \Illuminate\Http\Response
+     */
+    public function show(Team $team)
+    {
+        return view('manager.teams.show',compact('team'));
+    }
+
+    /**
+     * Show the form for editing the specified resource.
+     *
+     * @param  int  $id
+     * @return \Illuminate\Http\Response
+     */
+    public function edit(Team $team)
+    {
+        return view('manager.teams.edit',compact('team'));
+    }
+
+    /**
+     * Update the specified resource in storage.
+     *
+     * @param  \Illuminate\Http\Request  $request
+     * @param  int  $id
+     * @return \Illuminate\Http\Response
+     */
+    public function update(Request $request,Team $team)
+    {
+        $validated = $request->validate([
+            'name' => 'required|max:20', 
+        ]);
+        $team->update(["name"=>$request->name,
+        ]);
```

Comment:
1行にしてしまえるのでこれやっておきましょう。

```suggestion
        $team->update(["name"=>$request->name]);
```

Reply:
今後のために: この処理では特別問題ないのですが 以下のような感じのことも覚えておいてください。

$validated の変数にはチェック済みの値が入っているので、そちらを利用している方が安全なコードに見えます。

```php
$team->update(["name"=>$validated['name']]);
```

さらにこの場合、以下のようにもっとシンプルにも書けます。

```php
$team->update($validated);
```